### PR TITLE
Prepare for 1.29.0 release

### PR DESCRIPTION
**Pull request by jack-berg**

### CODE REVIEW

**Review by jkwatson — APPROVED** [review]
Thanks!

**Review by mateuszrzeszutek — APPROVED** [review]
👍

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,71 @@
...
+
+* Add Sdk{Signal}ProviderBuilder#addResource(Resource) method to merge Resource
+  with current
+  ([#5619](https://github.com/open-telemetry/opentelemetry-java/pull/5619))
+
+#### Metrics
+
+* Add LongHistogramAdviceConfigurer to improve api surface types
+  ([#5689](https://github.com/open-telemetry/opentelemetry-java/pull/5689))
+
```
Comment (trask) [inline CHANGELOG.md:20]:
```suggestion
* Increase metric name maximum length from 63 to 255 characters
  ([#5697](https://github.com/open-telemetry/opentelemetry-java/pull/5697))

```

Comment (jack-berg) [reply CHANGELOG.md:20]: We're going to be holding off on [#5697](https://github.com/open-telemetry/opentelemetry-java/pull/5697#issuecomment-1673753956) until next release.

**Pull request merged**